csrc/transform_iter.cpp

```diff
@@ -50,8 +50,14 @@ void ReplayTransformations::handle(Split* s) {
       "Transform traversal failed, modified a node but it was not a leaf node.");
 
   // Replay the split onto mapped
+  NVF_ERROR(s->outer()->isRFactorProduct() == s->inner()->isRFactorProduct());
   auto outs = IterDomain::split(
-      mapped, s->factor(), s->innerSplit(), s->startOffset(), s->stopOffset());
+      mapped,
+      s->factor(),
+      s->innerSplit(),
+      s->startOffset(),
+      s->stopOffset(),
+      replay_rfactor_ && s->outer()->isRFactorProduct());
```

Comment:
Suppose `mapped` is an rfactor product, this would result in an non-rfactor product output if `replay_rfactor_` is false. Should it be considered an error?

Reply:
> Suppose mapped is an rfactor product, this would result in an non-rfactor product output if replay_rfactor_ is false.

That's right. 

> Should it be considered an error?

That's already the current behavior and I'm not making it worse at least :)

AFAICT, it's not necessarily an error. Users of ReplayTransformations may want to replay rfactor-to-leaf transformations on a new rfactor domain. The results of those transformations shouldn't be marked `rf`. Correct?

Reply:
Below is a real scenario that happens ToT. It's not for split per say, but for merge, which I think is close enough to show the idea. It can be reproduced with `bin/nvfuser_tests --gtest_filter=*ForIncompliantLayout`. 

The pointwise scheduler tries to replay the consumer's root-to-leaf transformations on the producer's rfactor domain. Dimension 0 of the producer's rfactor doesn't have `rf` but dimension 1 has `rf`. 

![image](https://github.com/NVIDIA/Fuser/assets/2772612/02b0a224-edf7-4e65-86b8-87d3b5d00343)